src/app/application.cpp

```diff
@@ -449,14 +464,16 @@ void Application::runExternalProgram(const BitTorrent::Torrent *torrent) const
     });
     proc.startDetached();
 #else // Q_OS_WIN
-    // Cannot give users shell environment by default, as doing so could
-    // enable command injection via torrent name and other arguments
-    // (especially when some automated download mechanism has been setup).
-    // See: https://github.com/qbittorrent/qBittorrent/issues/10925
-    QStringList args = QProcess::splitCommand(program);
+    QStringList args = Utils::String::splitCommand(Preferences::instance()->getAutoRunProgram().trimmed());
```

Comment:
I don't see any obvious reasons not to make the code more unified, i.e. to make implementation of `splitCommand()` platform-dependent in order to use it under Windows as well.

Reply:
>to make implementation of splitCommand() platform-dependent in order to use it under Windows as well.

It is not that simple. `CommandLineToArgvW()` output is vastly different to `Utils::String::splitCommand()` so they definitely should not be combined together, otherwise unit testing code will be inconsistent and very weird.

Reply:
> IMO it shows that functions exists/serve for different reasons and shouldn't be combined.

OK, it looks like it's not quite what it seems at first glance. I'm sorry I caused you any trouble.

Reply:
>OK, it looks like it's not quite what it seems at first glance. I'm sorry I caused you any trouble.

It is ok.